luigi_monitor/luigi_monitor.py

```diff
@@ -8,17 +10,22 @@
 from luigi.retcodes import run_with_retcodes as run_luigi
 import requests
 
-events = {}
+const_success_message = "Job ran successfully!"
+
+
+class Monitor:
+    recorded_events = {}
```

Comment:
You could make this a `defaultdict(list)` (`from collections import defaultdict`) and save on all the `if 'Key' in m.recorded_events` checks (defaults to an empty list, so you can always call, for example `m.recorded_events['Missing'].append(task)`

Reply:
Alternatively, why not have separate properties (defaulting to empty lists) on `Monitor` for `missing`, `failure`, and `success`? That way they can be accessed as properties instead of w/ magic string keys.

(I have little idea where this fits in overall, but based on the changes in this PR, it looks like there's only going to be those 3 keys, so it seems like having them in a dict doesn't add anything)

Reply:
EDIT: actually read the whole file. 🔥  my last comment, but bump on the first one

Reply:
`defaultdict` is a great suggestions. Making the change. I think the other idea of separate properties could work but may need more refactoring.